app/Models/Collections/Artwork.php

```diff
@@ -335,6 +335,16 @@ public function webArtwork()
 
     // Meh, we'll leave out preferred & alternative places for now
 
+    public function imageEmbedding()
+    {
+        return $this->hasOne('App\Models\Web\Vectors\ImageEmbedding', 'model_id')->where('model_name', '=', 'artworks');
```

Comment:
Laravel provides built-in support for polymorphic relationships:
```php
return $this->morphOne(ImageEmbedding::class, 'image_embeddable', 'model_name', 'model_id');
```

See https://laravel.com/docs/11.x/eloquent-relationships#one-to-one-polymorphic-relations